traces/traces.go

```diff
@@ -0,0 +1,101 @@
+package traces
+
+import (
+	"context"
+	"fmt"
+	"runtime"
+
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/attribute"
+	semconv "go.opentelemetry.io/otel/semconv/v1.17.0"
+	"go.opentelemetry.io/otel/trace"
+)
+
+const (
+	instrumentationPkg = "github.com/osquery/osquery-go"
+	osqueryGoVersion   = "0.0.0"
+)
+
+// OsqueryGoTracer provides a tracer with a standardized name and version.
+func OsqueryGoTracer() trace.Tracer {
+	return otel.Tracer(instrumentationPkg, trace.WithInstrumentationVersion(osqueryGoVersion))
+}
+
+// StartSpan is a wrapper around trace.Tracer.Start that simplifies passing in span attributes.
+// `keyVals` should be a list of pairs, where the first in the pair is a string representing
+// the attribute key and the second in the pair is the attribute value.
+// The caller is always responsible for ending the returned span.
+// Any spans requiring more specific configuration can be created manually via OsqueryGoTracer().Start.
+func StartSpan(ctx context.Context, spanName string, keyVals ...interface{}) (context.Context, trace.Span) {
```

Comment:
I understand why we have the various bits of syntactic sugar here (and I love them in launcher) but I'm less sure about them in the. osquery SDK. Mostly, they're going to cause some work to happen even if spans are disabled. 

OTOH, runtime.Caller and some type switches are pretty cheap

Reply:
I went back and forth on this too. I think we probably wouldn't miss having e.g. the line numbers, so I'm fine with removing this for a lighter-touch integration if that's what folks would prefer.

Reply:
line numbers are probably not needed, there's so little code in osquery-go. So lighter might be simpler. I could be wrong though